[export] fix loss_output in joint graph signature

pianpwk opened the pull request:
Summary: joint-graph export is marking all outputs as LOSS_OUTPUT, fix so it marks only the correct one

Test Plan: test_experimental

Differential Revision: D66117412

ydwu4 commented on torch/export/graph_signature.py:
```diff
@@ -283,11 +283,15 @@ def user_inputs(self) -> Collection[Union[int, float, bool, None, str]]:
-            if s.kind != OutputKind.USER_OUTPUT:
+            if s.kind not in [
+                OutputKind.USER_OUTPUT,
+                OutputKind.LOSS_OUTPUT,
```
Why we need changing this?

pianpwk replied:
For verifier purposes this in a way includes the loss output: https://github.com/pytorch/pytorch/blob/d276688da6c1881d058670624fda4589f68b4571/torch/_export/verifier.py#L427

We don't have anything specific for training in the verifier yet.. ok maybe this is just a lazy change.

ydwu4 commented on test/export/test_experimental.py:
```diff
@@ -263,6 +263,26 @@ def forward(self, x, labels):
+        inputs = (torch.randn(4, 4),)
+        for i in [0, 1]:
+            ep = export_for_training(Foo(i), inputs)
+            ep0 = _export_forward_backward(ep, joint_loss_index=i)
```
also test the graph signatures's output spec?

pianpwk committed:
[export] fix loss_output in joint graph signature (#140974)

Summary:

joint-graph export is marking all outputs as LOSS_OUTPUT, fix so it marks only the correct one

Test Plan: test_experimental

Reviewed By: JacobSzwejbka

Differential Revision: D66117412
